invenio_circulation/transitions/transitions.py

```diff
@@ -149,22 +157,43 @@ def before(self, loan, **kwargs):
 
         self.ensure_item_is_available_for_checkout(loan)
 
-        no_pickup = not kwargs.get("pickup_location_pid") \
-            or "pickup_location_pid" not in loan
-        if no_pickup:
-            loan["pickup_location_pid"] = _get_item_location(loan["item_pid"])
+        _ensure_default_pickup_location(loan, kwargs)
 
         _ensure_valid_loan_duration(loan)
 
 
-class ItemAtDeskToItemOnLoan(ToItemOnLoan):
+class ItemAtDeskToItemOnLoan(Transition):
     """Check-out action to perform a loan when item ready at desk."""
 
     def before(self, loan, **kwargs):
         """Validate checkout action."""
         super().before(loan, **kwargs)
+
+        self.ensure_at_desk_item_is_available_for_checkout(loan)
+
+        _ensure_default_pickup_location(loan, kwargs)
+
         _ensure_valid_loan_duration(loan)
 
+    def ensure_at_desk_item_is_available_for_checkout(self, loan):
+        """Validate that an item at desk is available for checkout."""
+        self._check_item_before_availability(loan)
+
+        # patron_pid is mandatory for next steps
+        if 'patron_pid' not in loan:
+            msg = "Patron not set for loan with pid '{}'".format(loan['pid'])
+            raise TransitionConstraintsViolationError(description=msg)
+
+        # Some exceptions occurs with specific items and a given patron
```

Comment:
Very minor: probably the comment was more for the previous code, maybe you want to remove it or re-phrase it.

Reply:
You're right! This is not needed anymore.